x/distribution/keeper/allocation.go

```diff
@@ -48,10 +48,11 @@ func (k Keeper) AllocateTokens(ctx sdk.Context, sumPrecommitPower, totalPower in
 	for _, vote := range votes {
 		validator := k.stakingKeeper.ValidatorByConsAddr(ctx, vote.Validator.Address)
 
-		// TODO likely we should only reward validators who actually signed the block.
+		// TODO consider microslashing for missing votes.
 		// ref https://github.com/cosmos/cosmos-sdk/issues/2525#issuecomment-430838701
 		powerFraction := sdk.NewDec(vote.Validator.Power).Quo(sdk.NewDec(totalPower))
 		reward := feesCollected.MulDec(voteMultiplier).MulDec(powerFraction)
+		reward = reward.Cap(remaining)
```

Comment:
Hmm is this the best way to fix? If this is necessary, it seems like our multiplication somewhere rounds up where it ought to always floor, so instead I wonder if we should change the multiplication (`MulDec` maybe) to always floor, or add a version that does and use it here.